osh/cmd_eval.py

```diff
@@ -811,7 +811,8 @@ def _DoSimple(self, node, cmd_st):
             # shells aren't consistent.
             # self.mem.SetLastArgument('')
 
-        run_flags = executor.DO_FORK if node.do_fork else 0
+        # NOTE: ERR trap will not be called if not forked
+        run_flags = executor.DO_FORK if node.do_fork or self.trap_state.GetHook('ERR') else 0
```

Comment:
Hmm I'm still a bit confused by this one ...  also I'm not sure why test case 10 doesn't pass